### PR TITLE
chore(curriculum): add tests for globals in cash register

### PR DESCRIPTION
**Pull request by a2937**
Checklist:

- [X] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [X] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request).
- [X] My pull request targets the `main` branch of freeCodeCamp.
- [X] I have tested these changes either locally on my machine, or GitPod.

Related to #53836

This one adds two tests for the cash register. One for `cid` and another for `price`.  These tests will only work when the camper creates a `script` element on the page so the code for the page is actually linked.

### CODE REVIEW

**Comment by gikf** [comment]
Couldn't it be checked if `cid` and `price` are declared and can be reassigned instead - basically to try reassign them? The importance of their value before running tests is marginal, and campers often change these when manually testing.

**Comment by a2937** [comment]
I'm a little confused. Can you explain what you mean by check for reassignment? I know how to check if a variable is defined. That's easy enough. But I'm not entirely sure what I should be looking for. Also I think I found out that if you modify a variable in one test; the new value stays into the next one.

**Comment by gikf** [comment]
For example, for `price`:
```javascript
price = 10;
assert.strictEqual(price, 10);
```
If `price` is not declared, or declared with `const`, this will not pass.

**Comment by gikf** [comment]
Ouch, sorry, that's likely caused by user code being transformed, so `const` ends up transformed to `var`.

But `price` is then correctly being changed in further tests, isn't it?

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/build-a-cash-register-project/build-a-cash-register.md [hunk 1]
```diff
@@ -53,26 +53,18 @@ Fulfill the user stories and pass all the tests below to complete this project.
...
+You should have a variable called `price`. 
 
 ```js
-assert.strictEqual(price, 1.87);
+price = 10;
+assert.strictEqual(price, 10);
 ```
 
-You should leave the value of `cid` as `[["PENNY", 1.01],["NICKEL", 2.05],["DIME", 3.1],["QUARTER", 4.25],["ONE", 90],["FIVE", 55],["TEN", 20],["TWENTY", 60],["ONE HUNDRED", 100]]`
+You have a variable called `cid`.
```
Comment (ilenia-magoni) [inline curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/build-a-cash-register-project/build-a-cash-register.md:63]:
```suggestion
You should have a variable called `cid`. 
```

**Comment by ilenia-magoni** [comment]
> But `price` is then correctly being changed in further tests, isn't it?

I have seen it not pass with
```js
const price = ...
const cid = ...
```

but I don't know if it was only that or something else

**Comment by a2937** [comment]
> > But `price` is then correctly being changed in further tests, isn't it?
> 
> I have seen it not pass with
> 
> ```js
> const price = ...
> const cid = ...
> ```
> 
> but I don't know if it was only that or something else

The variables do get modified in at least a few tests.

https://github.com/freeCodeCamp/freeCodeCamp/blob/4b27d6597817fccc84433e02a53de4820956cc21/curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/build-a-cash-register-project/build-a-cash-register.md#L123-L134

**Comment by a2937** [comment]
> Ouch, sorry, that's likely caused by user code being transformed, so `const` ends up transformed to `var`.
> 
> But `price` is then correctly being changed in further tests, isn't it?

Yes. It is being modified in further tests. The solution still works even if `cid` and `price` are set as `const`.

**Comment by ilenia-magoni** [comment]
> > I have seen it not pass with
> > ```js
> > const price = ...
> > const cid = ...
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > but I don't know if it was only that or something else
> 
> The variables do get modified in at least a few tests.

Yeah, so I am confused on why I have seen it not pass, but your test doesn't work as it should

**Comment by a2937** [comment]
I don't know what to do here exactly. But it's essential that the `price` variable is modifiable for the other tests to pass. By some miracle the other tests aren't breaking. I don't think there's a single way to test the way we are with different cash drawer inputs and different prices; and have the two variables be unchanging. For now, whether it be bug or feature; it passes. My money is perhaps cached content. 

If that's the case, then we have the other tests to worry about.

**Comment by ilenia-magoni** [comment]
more than cached content I would guess it's the code being converted to `var`

**Comment by gikf** [comment]
I've checked with `var`, `let`, `const`, no issues with passing with otherwise good solution.

There's one case when tests (but all of them, not just new ones) confused the heck of me. I was starting with the seed code and I couldn't get the tests for global to pass `cid` and `price` are in the editor, but test are failing and console shows `RefferenceError`. The reason turned out simple - no html file with script linking the JS. This is kind of implied, but maybe it should be explicit.

**Comment by a2937** [comment]
Okay I have added a check to make sure that the html file is linking to script.js.

**Comment by ilenia-magoni** [comment]
the PR was a bit too behind, so I took the liberty to make a merge

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/build-a-cash-register-project/build-a-cash-register.md [hunk 2]
```diff
@@ -53,6 +53,27 @@ Fulfill the user stories and pass all the tests below to complete this project.
 
 # --hints--
 
+You should have the html file link to the Javascript
```
Comment (ilenia-magoni) [inline curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/build-a-cash-register-project/build-a-cash-register.md:56]:
```suggestion
You should have the HTML file link to the JavaScript file.
```

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/build-a-cash-register-project/build-a-cash-register.md [hunk 3]
```diff
@@ -53,6 +53,27 @@ Fulfill the user stories and pass all the tests below to complete this project.
...
+
+```js
+const script = document.querySelector('script[data-src$="script.js"]');
+assert.isNotNull(script); 
+```
+
+You should have a variable called `price`.
```
Comment (ilenia-magoni) [inline curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/build-a-cash-register-project/build-a-cash-register.md:63]:
```suggestion
You should have a global variable called `price`. 
```

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/build-a-cash-register-project/build-a-cash-register.md [hunk 4]
```diff
@@ -53,6 +53,27 @@ Fulfill the user stories and pass all the tests below to complete this project.
...
+```
+
+You should have a variable called `price`. 
+
+```js
+price = 10;
+assert.strictEqual(price, 10);
+```
+
+You should have a variable called `cid`.
```
Comment (ilenia-magoni) [inline curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/build-a-cash-register-project/build-a-cash-register.md:70]:
```suggestion
You should have a global variable called `cid`. 
```

**Review by majestic-owl448 — CHANGES_REQUESTED** [review]
The tests are good, I left a comment for the wording of the hints. After that, it's good to go.

**Review by majestic-owl448 — APPROVED** [review]

**Comment by a2937** [comment]
@gikf I've added the changes in #55195 into my PR.

**Review by gikf — APPROVED** [review]
LGTM 👍 

@a2937 do you want to add anything else here?

**Comment by a2937** [comment]
I can't think of anything else?

**Pull request merged**